stingray/largememory.py

```diff
@@ -156,15 +157,16 @@ def _combineSpectra(final_spectra):
     final_spectra.unnorm_power /= final_spectra.m
     # REVIEW: final_spectra.power_err /= final_spectra.m
 
-    if isinstance(final_spectra, stingray.AveragedPowerspectrum):
 
-        return final_spectra
+    # Don't look for pds1 and pds2 if it's a power spectrum.
+    # Since Powerspectrum is also a Crossspectrum, we need to specify this
```

Comment:
Doesn't AveragedPowerspectrum have a pds.power. Also if we use if and elif we don't need to check if the object is a Powerspectrum as well

Reply:
You were doing:
```
if Powerspectrum:
    return ...
elif Crossspectrum:
    pds1.power etcetc
    return ...
```

I condensed the two ifs into one.

Reply:
Ohh yes sorry my bad!